### PR TITLE
Extends spec coverage of Tax Rates settings - CRUD operations

### PR DESCRIPTION
**Pull request by filipefurtad0**
#### What? Why?

Related to #6016 and #6554.

Some CRUD operations were missing in the existing spec. This PR adds further test cases, like editing, deleting and verifying the "Included Price Validation".

#### Release notes

Added tests regarding editing fields, drop-downs and check-boxes and deleting a Tax Rate.
Added an assertion regarding the "Included Price Validation" for Tax Rates.

Changelog Category: Technical changes

No user facing changes.

### CODE REVIEW

**Review by andrewpbrett — APPROVED** [review]

File: spec/features/admin/configuration/tax_rates_spec.rb [hunk 1]
```diff
@@ -1,12 +1,12 @@
-# frozen_string_literal: true
```
Comment (Matt-Yorkley) [inline spec/features/admin/configuration/tax_rates_spec.rb:1]: Why was this removed?

File: spec/features/admin/configuration/tax_rates_spec.rb [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 require 'spec_helper'
 
 describe "Tax Rates" do
   include AuthenticationHelper
 
   let!(:calculator) { create(:calculator_per_item, calculable: create(:order)) }
-  let!(:tax_rate) { create(:tax_rate, calculator: calculator) }
+  let!(:tax_rate) { create(:tax_rate, name: "IVA", calculator: calculator) }
+  let!(:zone_id) { create(:zone, name: "Ilhas") }
+  let!(:tax_category_id) { create(:tax_category, name: "Full") }
```
Comment (Matt-Yorkley) [inline spec/features/admin/configuration/tax_rates_spec.rb:9]: I think these should be named `:zone` and `:tax_category`, as they are full objects and not just ids.

File: spec/features/admin/configuration/tax_rates_spec.rb [hunk 3]
```diff
@@ -29,4 +29,37 @@
...
+      uncheck("tax_rate[show_rate_in_label]")
+      check("tax_rate[included_in_price]")
+      click_button "Update"
+      expect(page).to have_content("Included Price Validation")
+    end
+
+    it "can be deleted", js: true do
+      click_link "Tax Rates"
+      accept_alert do
+        find(:xpath, "/html/body/div[1]/div[3]/div/div/div/table/tbody/tr[1]/td[8]/a[2]").click
```
Comment (Matt-Yorkley) [inline spec/features/admin/configuration/tax_rates_spec.rb:60]: This is pretty hard to read and understand (for anyone that has to come and change it in the future) and it's probably a bit fragile as well; it could break easily if the layout of the page changes at all, like for example if there's a slight change to the layout on admin pages, this spec could start failing.

We could do this, and it would delete the correct tax rate if there were multiple tax rates:
```ruby
find("#spree_tax_rate_#{tax_rate.id} .delete-resource").click
```
But in this test there's only ever one tax rate present, so we could even just do this:
```ruby
find(".delete-resource").click
```

Either one looks good to me. Much easier to read, and hopefully less prone to breaking if the layout changes slightly :ok_hand:

**Comment by Matt-Yorkley** [comment]
I left a couple of comments @filipefurtad0, but it's looking pretty good :+1:

File: spec/features/admin/configuration/tax_rates_spec.rb [hunk 4]
```diff
@@ -29,4 +29,37 @@
...
+    end
+
+    # See #6554: in order to set a Tax Rate as included in the price,
+    # there must be at least one Zone set the "Default Tax Zone"
+    it "checkboxes can be ticked" do
+      visit spree.edit_admin_tax_rate_path(tax_rate.id)
+      uncheck("tax_rate[show_rate_in_label]")
+      check("tax_rate[included_in_price]")
+      click_button "Update"
+      expect(page).to have_content("Included Price Validation")
```
Comment (Matt-Yorkley) [inline spec/features/admin/configuration/tax_rates_spec.rb:54]: This message will have changed now that https://github.com/openfoodfoundation/openfoodnetwork/pull/6545 has been merged, right?

File: spec/features/admin/configuration/tax_rates_spec.rb (hunk 1 above)
Comment (filipefurtad0) [reply spec/features/admin/configuration/tax_rates_spec.rb:1]: I deleted by mistake - I was not sure what it's doing, so I removed locally but ended up not adding it back; I'll return it there on the next commit.

File: spec/features/admin/configuration/tax_rates_spec.rb (hunk 4 above)
Comment (filipefurtad0) [reply spec/features/admin/configuration/tax_rates_spec.rb:55]: Yes, indeed it will - I'll add the correct string here :+1:

File: spec/features/admin/configuration/tax_rates_spec.rb (hunk 3 above)
Comment (filipefurtad0) [reply spec/features/admin/configuration/tax_rates_spec.rb]: This did look somewhat bulky/fragile and not readable. Thanks for suggesting this alternative!

**Review by Matt-Yorkley — APPROVED** [review]
Great :tada:

**Comment by sigmundpetersen** [comment]
Still a tax_rate related spec failing. Is it OK to merge or should this be looked at?

**Comment by filipefurtad0** [comment]
Thanks for spotting that @sigmundpetersen .
It's indeed related to a commit from this PR.
It passes locally every time, but if it introduces flakiness in the build, then maybe I should try to improve it?
If it breaks again, let's move it back to in dev :+1:

**Comment by Matt-Yorkley** [comment]
Looks like the new translation isn't being shown as expected in the test?

Screenshot from 2021-01-06 11-52-26

Possibly just needs a rebase on master..?

**Comment by filipefurtad0** [comment]
Hum, had a (first) attempt at rebasing this PR through git :-) Following this procedure:
https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request

First I did:
`git checkout master` and `git pull upstream master`

Then did:
i) `git rebase -i 9a199ae02c459a2898ead89bc7f3e25b99cf4ad5`
ii) Picked the first commit from the displayed file and squashed all others
iii) `git rebase upstream/master` -> no conflics
iv) `git push -f` and got:
image

The right code is now on my repo:
image

But I see still not on relevant branch for this PR:
image

Was this best approach? Any hints?

**Comment by Matt-Yorkley** [comment]
Not sure why the `git rebase -i 9a199ae02c459a2898ead89bc7f3e25b99cf4ad5` and other bits were needed...

Did you do `git checkout tax_rates_spec` and `git rebase master` afterwards? And then `git push origin tax_rates_spec -f`?

**Comment by filipefurtad0** [comment]
Yup, that worked :+1: 

I see I should to perform this operation on the branch to "be rebased" - missed that, and performed it in master. Also, I thought it would be best to do this in iterative mode, in case conflicts arise - but this is not expected here.

Thanks for your help on this Matt. I'll wait for the build to be green, and then move it to ready to go...

**Comment by Matt-Yorkley** [comment]
Hmmm... looks like it's all been squashed into one commit, but with a bunch of the changes missing :face_with_head_bandage:

**Comment by Matt-Yorkley** [comment]
I had just pulled a copy of your branch, about 10 seconds before you force-pushed. I can push it back if you like? It's got the other missing commits...

**Comment by filipefurtad0** [comment]
Sure, go ahead :+1:

**Comment by Matt-Yorkley** [comment]
Boom :boom: the missing commits are back :mage_man:

**Comment by filipefurtad0** [comment]
Ok, so the failing spec is probably a flaky one - the one breaking it before is now fixed.

So, I guess my homework is trying to figure out what went wrong on my rebasing-attempt :sweat_smile: 
Thanks again :+1:

**Comment by Matt-Yorkley** [comment]
All green :heavy_check_mark:
